### ISSUE TITLE
assertSnapshot does not fail if view's size is changed to zero

### ISSUE THREAD

**Issue by krzysztofpawski** [pointfreeco/swift-snapshot-testing#414]
We found out that we had a failing test in our repo for two weeks but we didn't notice it because tests were passing even if they shouldn't be. The source of the not passing tests was our incorrect input data but still it's a pitty that `SnapshotTesting` didn't warn us about it.

Start with such test
```
func testExample() throws {
    isRecording = true
    let view = UIView()
    view.backgroundColor = .blue
    view.frame = CGRect(x: 0, y: 0, width: 200, height: 400)
    assertSnapshot(matching: view, as: .image)
}
```
It should record the blue rectangle.
Now change the test so we assert against empty view
```
func testExample() throws {
    let view = UIView()
    assertSnapshot(matching: view, as: .image)
}
```
And try this:
```
func testExample() throws {
    isRecording = true
    let view = UIView()
    assertSnapshot(matching: view, as: .image)
}
```
**Result:**
With second attempt test is passing.
With third attempt "empty" view is recorded.

**Expected result:**
With second attempt test is failing.

**Comment by nuno-vieira** [comment]
The suggested fix is here btw: https://github.com/pointfreeco/swift-snapshot-testing/issues/414 @stephencelis feel free to provide feedback or a different solution if you think there is a better one 👍

**Comment by stephencelis** [comment]
Fixed by #453.
